src/explore-education-statistics-common/src/modules/table-tool/utils/createExpandedRowHeaders.ts

```diff
@@ -42,21 +46,34 @@ export default function createExpandedRowHeaders(
         !matchesPreviousHeader ||
         (matchesPreviousHeader && current.hasSiblings())
       ) {
-        const isGroup = current.hasChildren();
+        // Check to see if all headers at the current level have only
+        // a single matching child so will be merged.
+        const headersAtCurrentLevel = current.parent
+          ? current.parent.children
+          : rowHeaders;
+        const allHeadersAtCurrentLevelWillBeMerged = headersAtCurrentLevel.every(
+          rowHeader => rowHeader.hasSingleMatchingChild(),
+        );
```

Comment:
Similarly to `createExpandedColumnHeaders`, this is a little inefficient as we're having to re-calculate this every single time we push a header to the row, so we're potentially repeating ourselves.

Seems like it would be fairly easy to fix this with something like:

```ts
export default function createExpandedRowHeaders(
  rowHeaders: Header[],
): TableCellJson[][] {
  // ...

  // Keep track of levels that can be collapsed (i.e. when each
  // header in the level only has a single matching child)
  let collapsibleLevels: boolean[] = [
    rowHeaders.every(rowHeader => rowHeader.hasSingleMatchingChild()),
  ];

  return rowHeaders.reduce<TableCellJson[][]>((acc, header) => {
    // ...

    // 👇️ Keep track of the depths that have been checked for 
    const depthsChecked = new Set<number>();

    while (stack.length > 0) {
      // ...

      if (
        !matchesPreviousHeader ||
        (matchesPreviousHeader && current.hasSiblings())
      ) {
        const { depth, crossSpan } = current;

        if (collapsibleLevels[depth] === undefined && current.parent) {
          collapsibleLevels[depth] = current.parent.children.every(rowHeader =>
            rowHeader.hasSingleMatchingChild(),
          );
        }

        row.push({
          text: current.text,
          rowSpan: current.span,
          scope: maxDepth > depth + crossSpan ? 'rowgroup' : 'row',
          colSpan: collapsibleLevels[depth] ? 1 : crossSpan,
          tag: 'th',
        });
      }

      // ...
    }

    // 👇️ Need to remember to reset this before moving to next top-level header
    collapsibleLevels = [collapsibleLevels[0]];

    return acc;
  }, []);
}
```

As a side benefit, this actually solves another test case I found by tweaking `createExpandedRowHeaders > returns correct headers with deeply nested rows and multiple identical headers` a little with the following headers:

```ts
const rowHeaders: Header[] = [
  new Header('A', 'A').addChild(
    new Header('A', 'A').addChild(
      new Header('A', 'A').addChild(
        new Header('A', 'A').addChild(new Header('A', 'A')),
      ),
    ),
  ),
  new Header('B', 'B').addChild(
    new Header('B', 'B')
      .addChild(
        new Header('B', 'B').addChild(
          new Header('B', 'B').addChild(new Header('B', 'B')),
        ),
      )
      .addChild(
        new Header('C', 'C').addChild(
          new Header('D', 'D').addChild(new Header('D1', 'D1')),
        ),
      ),
  ),
  new Header('E', 'E').addChild(
    new Header('F', 'F')
      .addChild(
        new Header('F', 'F')
          .addChild(new Header('F', 'F').addChild(new Header('F', 'F')))
          .addChild(new Header('G', 'G').addChild(new Header('G1', 'G1'))),
      )
      .addChild(
        new Header('H', 'H')
          .addChild(new Header('I', 'I').addChild(new Header('I', 'I')))
          .addChild(new Header('J', 'J').addChild(new Header('J', 'J'))),
      ),
  ),
];
```

With the current implementation:

![image](https://github.com/dfe-analytical-services/explore-education-statistics/assets/9917868/cbe40705-72ca-4fb7-9637-d14533e63065)

With the new level tracking:
![image](https://github.com/dfe-analytical-services/explore-education-statistics/assets/9917868/a6476930-a9d3-45b9-91bc-35d6068e06e7)

Probably worth adding this test case in (or changing the existing one).